### PR TITLE
Add `@s3UnwrappedXmlOutput` trait to support S3's GetBucketLocation

### PR DESCRIPTION
**Pull request by jdisanti**
This PR adds a new `@s3UnwrappedXmlOutput` trait so that the response shape for S3's GetBucketLocation call can be expressed in the Smithy model, and also adds an S3 protocol test to exercise the new trait.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by kggilmer** [comment]
imho It seems a little weird to create a trait to fix a specific problem on a specific operation, as one would expect traits to be more general purpose.  I'm guessing that this is the approach because, due to the structure of the response, Smithy is unable to describe it.  Would a more general-purpose approach be a trait which would collapse the wrapper of any XML structure/shape, similar to how maps and lists can be flattened?

**Review by kstich — CHANGES_REQUESTED** [review]

File: smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/customizations/S3UnwrappedXmlOutputTrait.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+package software.amazon.smithy.aws.traits.customizations;
+
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.AnnotationTrait;
+
+public final class S3UnwrappedXmlOutputTrait extends AnnotationTrait {
+    public static final ShapeId ID = ShapeId.from("smithy.api#s3UnwrappedXmlOutput");
```
[the remark this answers is not in the document]
Comment (jdisanti) [reply smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/customizations/S3UnwrappedXmlOutputTrait.java:24]: Good catch! Fixed.

File: docs/source/1.0/spec/aws/customizations/s3-customizations.rst [hunk 2]
```diff
@@ -130,3 +130,22 @@ resolved to "virtual" to enable this setting.
 .. _path-style requests: https://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html#path-style-access
 .. _dual-stack endpoints: https://docs.aws.amazon.com/AmazonS3/latest/dev/dual-stack-endpoints.html
 .. _transfer acceleration: https://docs.aws.amazon.com/AmazonS3/latest/dev/transfer-acceleration.html
+
+S3 Unwrapped XML Responses
+--------------------------
```
[the remark this answers is not in the document]
Comment (jdisanti) [reply docs/source/1.0/spec/aws/customizations/s3-customizations.rst:135]: Ah, I didn't notice that. Revised the documentation to match the other trait docs and gave it its own top-level section. Thanks!

**Comment by jdisanti** [comment]
@kggilmer - The intent is to be able to express S3 in Smithy while avoiding a more generic mechanism that would allow future services to have unwrapped responses like this.

**Review by kstich — APPROVED** [review]

**Review by JordonPhillips — APPROVED** [review]

**Pull request merged**